src/compiler.rs

```diff
@@ -220,22 +214,23 @@ impl AmberCompiler {
                 dep_path
             };
             let document = block.document(&meta);
-            // Save to file
+            // Save to file; replace the base directory if the output
+            // path is absolute, otherwise append the output path.
```

Comment:
I think that when you need to leave comments to know what you're doing the code is too complex. This function is very long and creates many variables. Can you try to separate it to smaller helper functions?

Reply:
While I'm normally all for refactoring to improve the code, there is another outstanding pull request #465 which modifies this file as well.  I'm already going to have to fix merge conflicts after that one is merged, and I don't want to make extra work for myself.  Perhaps we can revisit this another time?